ground/gcs/src/plugins/boards_taulabs/sparky.cpp

```diff
@@ -174,16 +181,46 @@ enum Core::IBoardType::InputType Sparky::getInputType()
     case HwSparky::RCVRPORT_PPM:
         return INPUT_TYPE_PPM;
     case HwSparky::RCVRPORT_SBUS:
-	    return INPUT_TYPE_SBUS;
+        return INPUT_TYPE_SBUS;
     case HwSparky::RCVRPORT_SBUSNONINVERTED:
-		return INPUT_TYPE_SBUSNONINVERTED;
-	case HwSparky::RCVRPORT_DSM:
+        return INPUT_TYPE_SBUSNONINVERTED;
+    case HwSparky::RCVRPORT_DSM:
         return INPUT_TYPE_DSM;
     case HwSparky::RCVRPORT_HOTTSUMD:
         return INPUT_TYPE_HOTTSUMD;
-    default:
-        return INPUT_TYPE_UNKNOWN;
+    case HwSparky::RCVRPORT_HOTTSUMH:
+        return INPUT_TYPE_HOTTSUMH;
+    case HwSparky::RCVRPORT_IBUS:
+        return INPUT_TYPE_IBUS;
+    case HwSparky::RCVRPORT_SRXL:
+        return INPUT_TYPE_SRXL;
+    }
+    
+    switch(settings.FlexiPort) {
+    case HwSparky::FLEXIPORT_DSM:
+        return INPUT_TYPE_DSM;
+    case HwSparky::FLEXIPORT_SBUS:
+        return INPUT_TYPE_SBUS;
+    case HwSparky::FLEXIPORT_SBUSNONINVERTED:
+        return INPUT_TYPE_SBUSNONINVERTED;
+    case HwSparky::FLEXIPORT_IBUS:
+        return INPUT_TYPE_IBUS;
+    case HwSparky::FLEXIPORT_SRXL:
+        return INPUT_TYPE_SRXL;
```

Comment:
default: break; is needed on all switch statements that are not expected to cover all possible values in an enum